### ISSUE TITLE
[filter-effects-1] Filter primitives do not always form a tree structure

### ISSUE THREAD

**Issue by jarek-foksa** [w3c/fxtf-drafts#376]
[Section 9.3](https://drafts.fxtf.org/filter-effects-1/#FilterPrimitiveTree) states that every filter structure can be represented as a tree. From my testing this statement seems to be wrong as it is possible to organize the primitives so that they form a more generic graph rather than a tree. For example:

```xml
<filter id="test-filter">
  <feFlood result="red-flood" flood-color="red" flood-opacity="0.5" />
  <feFlood result="green-flood" flood-color="green" flood-opacity="0.5" />
  <feMerge result="red-greeen-flood">
    <feMergeNode in="red-flood"/>
    <feMergeNode in="green-flood"/>
  </feMerge>
  <feMerge>
    <feMergeNode in="red-green-flood"/>
    <feMergeNode in="green-flood"/>
  </feMerge>
</filter>
```

<img width="300" alt="Screen Shot 2019-09-23 at 20 01 01" src="https://user-images.githubusercontent.com/193371/65451067-bc4cd000-de3e-11e9-9dce-2caa376b68ad.png">

**Comment by dirkschulze** [comment]
@jarek-foksa Stricktly speaking, according to the graph theory, a filter chain does not suffice the requirements of a tree (one vertex is connected to another by exactly one path).  It does seem to fulfill the requirements of a [directed acyclic graph](https://en.m.wikipedia.org/wiki/Directed_acyclic_graph), a polytree.

**Comment by jarek-foksa** [comment]
Yeah, I would adjust the spec to state that the filter primitives represent a directed acyclic graph, which in turn could be represented by an equivalent tree graph.

The current wording suggests that you can organize primitives only in a tree structure.
